client/src/test/java/io/pravega/client/netty/impl/ConnectionFactoryImplTest.java

```diff
@@ -0,0 +1,259 @@
+
+public class ConnectionFactoryImplTest {
+
+    boolean ssl = false;
+    private Channel serverChannel;
+    private int port;
+
+    @Before
+    public void setUp() throws Exception {
+        // Configure SSL.
+        port = TestUtils.getAvailableListenPort();
+        final SslContext sslCtx;
+        if (ssl) {
+            try {
+                sslCtx = SslContextBuilder.forServer(new File("../config/cert.pem"), new File("../config/key.pem")).build();
+            } catch (SSLException e) {
+                throw new RuntimeException(e);
+            }
+        } else {
+            sslCtx = null;
+        }
+        boolean nio = false;
+        EventLoopGroup bossGroup;
+        EventLoopGroup workerGroup;
+        try {
+            bossGroup = new EpollEventLoopGroup(1);
+            workerGroup = new EpollEventLoopGroup();
+        } catch (ExceptionInInitializerError | UnsatisfiedLinkError | NoClassDefFoundError e) {
+            nio = true;
+            bossGroup = new NioEventLoopGroup(1);
+            workerGroup = new NioEventLoopGroup();
+        }
+
+        ServerBootstrap b = new ServerBootstrap();
+        b.group(bossGroup, workerGroup)
+         .channel(nio ? NioServerSocketChannel.class : EpollServerSocketChannel.class)
+         .option(ChannelOption.SO_BACKLOG, 100)
+         .handler(new LoggingHandler(LogLevel.INFO))
+         .childHandler(new ChannelInitializer<SocketChannel>() {
+             @Override
+             public void initChannel(SocketChannel ch) throws Exception {
+                 ChannelPipeline p = ch.pipeline();
+                 if (sslCtx != null) {
+                     SslHandler handler = sslCtx.newHandler(ch.alloc());
+                     SSLEngine sslEngine = handler.engine();
+                     SSLParameters sslParameters = sslEngine.getSSLParameters();
+                     sslParameters.setEndpointIdentificationAlgorithm("LDAPS");
+                     sslEngine.setSSLParameters(sslParameters);
+                     p.addLast(handler);
+                 }
+             }
+         });
+
+        // Start the server.
+        serverChannel = b.bind("localhost", port).awaitUninterruptibly().channel();
+    }
+
+    @After
+    public void tearDown() throws Exception {
+        serverChannel.close();
+        serverChannel.closeFuture();
+    }
+
+    @Test
+    public void establishConnection() throws ConnectionFailedException {
+        ConnectionFactoryImpl factory = new ConnectionFactoryImpl(PravegaClientConfig.builder()
+                                    .controllerURI(URI.create((this.ssl ? "tls://" : "tcp://") + "localhost"))
+                                    .pravegaTrustStore("../config/cert.pem")
+                                   .build());
+        ClientConnection connection = factory.establishConnection(new PravegaNodeUri("localhost", port), new ReplyProcessor() {
+            @Override
+            public void hello(WireCommands.Hello hello) {
```

Comment:
Use FailingReplyProcessor to avoid this. Here and elsewhere.

Reply:
Fixed.